### DISCUSSION TITLE
<template x-if='...'> requires <div> inside form

### DISCUSSION THREAD

**Discussion by ronaaron** [alpinejs/alpine#3826] [7. Bugs (archived)] [answered]
I have a form where I present fields based on selections made in the form. To do that, I am using a template with x-if. That works if I have a single form element, but if I want to display more than one element, only the first one is displayed. If I wrap the template contents in a div, it works (that is, everything in the div is shown).

I was hoping I wouldn't have to do it this way.

Example that doesn't work:
~~~
<template x-if='show'>
   <button>something</button>
   <label>just text</label>
   <input x-model='justtext' />
</template>
~~~

Example that does work:
~~~
<template x-if='show'>
  <div>
   <button>something</button>
   <label>just text</label>
   <input x-model='justtext' />
  </div>
</template>
~~~

**Comment by browner12** [comment, marked as answer]
According to the [docs](https://alpinejs.dev/directives/if):

> Remember: \<template\> tags can only contain one root level element.

**Comment by ronaaron** [comment]
Oof. I didn't notice that in the documentation. It would be *nice* if it worked as I described, though...

**Reply by jablonski** [reply]
You can use an empty fragment so you don't have to change your css:

```
<template x-if='show'>
  <>
   <button>something</button>
   <label>just text</label>
   <input x-model='justtext' />
  </>
</template>
```

**Reply by ekwoka** [reply]
`<>` isn't html.

You instead can use a span with `display: content`

**Comment by ronaaron** [comment]
Nice, thanks!

**Reply by ekwoka** [reply]
you can use a span/div with `display: content` to wrap the multiple children. content make the element act like a passthrough for css and layout, while still wrapping the nodes in the dom tree for Alpine to work with.

**Comment by ronaaron** [comment]
I'll just reiterate my hope that Alpine could treat any template as if whatever it contained was a single unit. Perhaps by implicitly wrapping the  contents as mentioned.

**Reply by ekwoka** [reply]
this is a PR I have planned after a few others.

Would just need to do some testing to be sure it wouldn't normally mess with any css/layout/aria stuff
